lib/get_text_command.dart

```diff
@@ -0,0 +1,212 @@
+          final matchingJson = json.decode(finderData['matching'] as String);
+
+          return Ancestor(
+            of: deserializeBase64(Base64URL.encode(json.encode(ofJson))),
+            matching:
+                deserializeBase64(Base64URL.encode(json.encode(matchingJson))),
+            matchRoot: finderData['matchRoot'] == 'true',
+            firstMatchOnly: finderData['firstMatchOnly'] == 'true',
+          );
+
+        case 'Descendant':
+          final ofJson = json.decode(finderData['of'] as String);
+          final matchingJson = json.decode(finderData['matching'] as String);
+
+          return Descendant(
+            of: deserializeBase64(Base64URL.encode(json.encode(ofJson))),
+            matching:
+                deserializeBase64(Base64URL.encode(json.encode(matchingJson))),
+            matchRoot: finderData['matchRoot'] == 'true',
+            firstMatchOnly: finderData['firstMatchOnly'] == 'true',
+          );
+
+        default:
+          throw Exception('Unsupported finder type: $finderType');
+      }
+    } catch (e) {
+      throw Exception('Error deserializing finder: $e');
+    }
+  }
+}
+
+class GetTextCommandExtension extends CommandExtension {
+  String getTextFromWidget(Text widget) {
+    // If direct text data is available, return it
+    if (widget.data != null) {
+      return widget.data!;
+    }
+    // Otherwise get text from textSpan
+    if (widget.textSpan != null) {
+      return getTextFromTextSpan(widget.textSpan!);
+    }
+    return 'TEXT_NOT_FOUND';
+  }
+
+  String getTextFromTextSpan(InlineSpan span) {
+    if (span is TextSpan) {
+      final List<String> textParts = <String>[];
+
+      // Add current text if not null
+      if (span.text != null) {
+        textParts.add(span.text!);
+      }
+
+      // Add children's text recursively
+      if (span.children != null) {
+        for (final InlineSpan child in span.children!) {
+          textParts.add(getTextFromTextSpan(child));
+        }
+      }
+
+      return textParts.join('');
+    }
+    return 'WIDGET_IS_NOT_TEXT_SPAN';
+  }
+
+  @override
+  Future<Result> call(
+      Command command,
+      WidgetController prober,
+      CreateFinderFactory finderFactory,
+      CommandHandlerFactory handlerFactory) async {
+    final GetTextCommand dragCommand = command as GetTextCommand;
+
+    // Create finder for Text widget
+    final type = dragCommand.base64Element;
+    // decodeBase64 to json
+    SerializableFinder serializableFinder =
+        FinderHelper.deserializeBase64(type);
+
+    final Finder finder = finderFactory.createFinder(serializableFinder);
+
+    // Get the widget element
+    final Element element = prober.element(finder);
+
+    // Get text data from Text widget
+    String textData = '';
+
+    if (element.widget is Text) {
+      textData = getTextFromWidget(element.widget as Text);
+    } else {
+      textData = 'Found element is not a Text widget';
```

Comment:
Returning an error in the response for the user could be a potential improvement, making the behavior clearer and more understandable for others.

Example code:
```
      return const GetTextResult(
        false,
        data: {
          'errorCode': 'NOT_A_TEXT_WIDGET',
          'error': 'Found element is not a Text widget'
        },
      );
```
Response if the request is incorrect:
```
{'isError': false, 'response': {'success': false, 'errorCode': 'NOT_A_TEXT_WIDGET', 'error': 'Found element is not a Text widget'}, 'type': '_extensionType', 'method': 'ext.flutter.driver'}
```

At the moment, we are receiving this:
```
{'isError': false, 'response': {'success': true, 'text': 'Found element is not a Text widget'}, ...}
```

Reply:
nice. I updated the code handling get text error, and rename the drag command as your suggestion [here](https://github.com/appium/appium-flutter-driver/pull/755#discussion_r1864802802)

Reply:
looks great, thank you so much!